mono_repo/lib/src/presubmit_command.dart

```diff
@@ -0,0 +1,128 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:args/command_runner.dart';
+import 'package:io/ansi.dart';
+import 'package:path/path.dart' as p;
+import 'package:pub_semver/pub_semver.dart';
+
+import 'travis_config.dart';
+import 'utils.dart';
+
+class PresubmitCommand extends Command {
+  @override
+  String get name => 'presubmit';
+
+  @override
+  String get description => 'Run the travis presubmits locally.';
+
+  PresubmitCommand() {
+    argParser.addOption('package',
+        help: 'The package(s) to run on, defaults to all packages',
+        allowMultiple: true,
+        abbr: 'p');
+    argParser.addOption('task',
+        help: 'The task(s) to run, defaults to all tasks',
+        allowMultiple: true,
+        abbr: 't');
+    argParser.addOption('sdk',
+        help: 'Which sdk to use for match tasks, defaults to current sdk',
+        allowMultiple: false,
+        defaultsTo: _currentSdk,
+        allowed: ['dev', 'stable']);
+  }
+
+  @override
+  Future run() async {
+    var passed = await presubmit(
+        packages: argResults['package'] as List<String>,
+        tasks: argResults['task'] as List<String>,
+        sdkToRun: argResults['sdk'] as String);
+
+    // Set a bad exit code if it failed.
+    if (!passed) exitCode = 1;
+  }
+}
+
+final _currentSdk =
+    new Version.parse(Platform.version.split(' ').first).isPreRelease
+        ? 'dev'
+        : 'stable';
+
+/// A temp dir to dump logs to for errors.
+Directory __tmpDir;
+Directory get _tmpDir => __tmpDir ??= Directory.systemTemp.createTempSync();
```

Comment:
Gah! Top-level mutable state.

Couldn't this go inside the `presubmit` function itself?

Reply:
Done.